haiku.py

```diff
@@ -0,0 +1,178 @@
+'''
```

Comment:
I sourced this file from [here](https://github.com/odemiral/Haiku-Bot).

Reply:
Let's link to the original in the comments at least. So we a) know where to find it in the future and b) are giving appropriate credit. We could *technically* do this with submodules, but if we're just using a single file, that's a bit overkill.